contracts/strategies/UniversalLendStrategy.sol

```diff
@@ -198,27 +198,29 @@ abstract contract UniversalLendStrategy is ProxyStrategyBase {
       require(_localBalance < _DUST || profit < poolBalance / 20, 'Too huge profit');
 
       uint toBuybacks = _calcToBuyback(profit, _localBalance);
-      uint remaining = profit - toBuybacks;
-      if (remaining != 0) {
-        localBalance += remaining;
-      }
+
       if (toBuybacks > _DUST) {
         // if no users, withdraw all and send to controller for remove dust from this contract
         if (toBuybacks == poolBalance) {
           _withdrawAllFromPool();
           localBalance = 0;
           IERC20(u).safeTransfer(address(c), IERC20(u).balanceOf(address(this)));
         } else {
-          bool withdrewAll = _withdrawFromPoolWithoutChangeLocalBalance(toBuybacks, poolBalance);
+          (bool withdrewAll, uint withdrawnAmount) = _withdrawFromPoolWithoutChangeLocalBalance(toBuybacks, poolBalance);
           if (withdrewAll) {
             localBalance = 0;
           }
-          IERC20(u).safeTransfer(_PERF_FEE_TREASURY, toBuybacks);
+          IERC20(u).safeTransfer(_PERF_FEE_TREASURY, withdrawnAmount);
+          uint remaining = profit - withdrawnAmount; // need to use real withdrawn amount instead of toBuybacks
+          if (remaining != 0) {
+            localBalance += remaining;
+          }
         }
       }
+
     }
     IBookkeeper(c.bookkeeper()).registerStrategyEarned(0);
-    return _getActualPoolBalance();
+    return poolBalance;
```

Comment:
why?

Reply:
You are correct. It shouldn't be changed. Reverted.